experimental/AffinePlaneCurve.jl

```diff
@@ -0,0 +1,355 @@
+  return !(iszero(a) && iszero(b))
+end
+
+################################################################################
+# If P is a smooth point of an affine plane curve C, compute the tangent of C
+# at the point P.
+
+@doc Markdown.doc"""
+    tangent(C::AffinePlaneCurve{S}, P::Point{S}) where S <: FieldElem
+
+Return the tangent of `C` at `P` when `P` is a smooth point of `C`, and throw an error otherwise.
+"""
+function tangent(C::AffinePlaneCurve{S}, P::Point{S}) where S <: FieldElem
+  P.ambient_dim == 2 || error("The point needs to be in a two dimensional space")
+  iszero(evaluate(C.eq, P.coord)) || error("The point is not on the curve")
+  J = jacobi_ideal(C)
+  L = gens(J)
+  a = evaluate(L[1], P.coord)
+  b = evaluate(L[2], P.coord)
+  if iszero(a) && iszero(b)
+     error("This is a singular point of the curve")
+  else
+     V = gens(parent(C.eq))
+     T = a*(V[1] - P.coord[1]) + b*(V[2] - P.coord[2])
+     return AffinePlaneCurve(T)
+  end
+end
+
+################################################################################
+# gives the common components of two affine plane curves
+
+@doc Markdown.doc"""
+    common_components(C::AffinePlaneCurve{S}, D::AffinePlaneCurve{S}) where S <: FieldElem(C::AffinePlaneCurve)
+
+Return the affine plane curve consisting of the common component of `C` and `D`, or an empty vector if they do not have a common component.
+"""
+function common_components(C::AffinePlaneCurve{S}, D::AffinePlaneCurve{S}) where S <: FieldElem
+  G = gcd(C.eq, D.eq)
+  if isone(G)
+     return Vector{AffinePlaneCurve}()
+  else
+     return [AffinePlaneCurve(G)]
+  end
+end
+
+################################################################################
+# Compute the intersection of two curves. The output is a list: the first
+# element the affine plane curve defined by the common components of the two
+# curves (or is empty if no common component), the second element is the list
+# of intersection points. Some might be contained in the common component too.
+
+@doc Markdown.doc"""
+    curve_intersect(C::AffinePlaneCurve{S}, D::AffinePlaneCurve{S}) where S <: FieldElem
+
+Return a list whose first element is the affine plane curve defined by the gcd of `C.eq` and `D.eq`, the second element is the list of the remaining intersection points when the common components are removed from `C` and `D`.
+"""
+function curve_intersect(C::AffinePlaneCurve{S}, D::AffinePlaneCurve{S}) where S <: FieldElem
+  G = gcd(C.eq, D.eq)
+  R = parent(C.eq)
+  if !isone(G)
+     # We divide by the gcd to get curves without common components.
+     F = div(C.eq, G)
+     H = div(D.eq, G)
+  else
+     F = C.eq
+     H = D.eq
+  end
+  I = ideal(R, [F, H])
+  # We eliminate the first variable in the ideal.
+  B = eliminate(I, [gen(R, 1)])
+  # We factorize the polynomial we obtain (which is a polynomial where only
+  # the second variable appears).
+  Z = factor(gen(B, 1))
+  Y = []
+  L = Array{S, 1}[]
```

Comment:
Perhaps like this, for consistency?
```suggestion
  L = Vector{S}[]
```